lib/src/view/user/game_history_screen.dart

```diff
@@ -1,65 +1,130 @@
+import 'package:chessground/chessground.dart';
 import 'package:flutter/cupertino.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:lichess_mobile/src/model/auth/auth_session.dart';
 import 'package:lichess_mobile/src/model/common/perf.dart';
+import 'package:lichess_mobile/src/model/game/game_filter.dart';
 import 'package:lichess_mobile/src/model/game/game_history.dart';
 import 'package:lichess_mobile/src/model/user/user.dart';
+import 'package:lichess_mobile/src/model/user/user_repository_providers.dart';
+import 'package:lichess_mobile/src/styles/lichess_icons.dart';
 import 'package:lichess_mobile/src/utils/l10n_context.dart';
 import 'package:lichess_mobile/src/view/game/game_list_tile.dart';
+import 'package:lichess_mobile/src/widgets/adaptive_bottom_sheet.dart';
 import 'package:lichess_mobile/src/widgets/feedback.dart';
-import 'package:lichess_mobile/src/widgets/platform.dart';
 
 class GameHistoryScreen extends ConsumerWidget {
   const GameHistoryScreen({
     required this.user,
     required this.isOnline,
-    this.perf,
-    this.games,
+    this.gameFilter = const GameFilterState(),
     super.key,
   });
   final LightUser? user;
   final bool isOnline;
-  final Perf? perf;
-  final int? games;
+  final GameFilterState gameFilter;
 
   @override
   Widget build(BuildContext context, WidgetRef ref) {
-    return ConsumerPlatformWidget(
-      ref: ref,
-      androidBuilder: _buildAndroid,
-      iosBuilder: _buildIos,
+    final session = ref.read(authSessionProvider);
+    final username = user?.name ?? session?.user.name;
+    final title = Text(
+      username != null
+          ? '$username ${context.l10n.games.toLowerCase()}'
+          : context.l10n.games,
     );
+    final filtersInUse = ref
```

Comment:
Cf. other comment. `ref.read` should not be used in `build()` methods; to count the filters you should get the filter state with `ref.watch`.